test/end-to-end/auth-client.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const _ = require('lodash');
+const nightwatch = require('nightwatch');
+const test = require('ava');
+
+const smartcar = require('../../');
+
+const {getAuthClientParams, runAuthFlow} = require('./helpers');
+const config = require('../config');
+
+const context = {};
+
+test.before(() => {
+  context.client = nightwatch.initClient(config.get('nightwatch'));
+  context.browser = context.client.api();
+});
+
+test('exchangeCode', (t) => {
+  const client = new smartcar.AuthClient(getAuthClientParams());
+
+  const authUrl = client.getAuthUrl();
+
+  return runAuthFlow(context.client, context.browser, authUrl)
+    .then(async (code) => {
+      try {
+        const access = await client.exchangeCode(code);
+        const keys = Object.keys(access).sort();
+        const hasKeys = _.isEmpty(
+          _.xor(keys, [
+            'accessToken',
+            'expiration',
+            'refreshExpiration',
+            'refreshToken',
+          ])
+        );
+        t.true(hasKeys);
+
+        t.pass();
+      } catch (err) {
+        t.fail();
```

Comment:
you should just let the error bubble up, this makes the test harder to debug, no?